woocommerce/storefront
# Update dependencies and fix e2e tests

This PR:

* Bumps webpack from 5.831.1 to 5.94.0 (same as PR https://github.com/woocommerce/storefront/pull/2159).
* Updates the version of Node to be in sync with WooCommerce.
* I included `validate-build.js` to the excluded files from builds.
* Updated several other packages that were necessary. That also involved:
  * Disabling a new ESLint rule: `@typescript-eslint/no-this-alias`.
  * Updating the only e2e test we have.
  * Rebuilding `package-lock.json` from scratch.

This allows us to get the e2e tests job pass again.

### How to test the changes in this Pull Request:

This PR doesn't include any new features, it's simply updating dependencies and internal tools.

1. Run `npm ci && npm run build` to build the ZIP.
2. Install the generated ZIP in a fresh store.
3. Do some smoke testing.
4. Verify everything works as it used to and there are no regressions.

### Changelog

> Update many dev dependencies.
